internal/controllers/gateway/gateway_controller.go

```diff
@@ -48,9 +48,9 @@ type GatewayReconciler struct { //nolint:revive,golint
 	Scheme          *runtime.Scheme
 	DataplaneClient *dataplane.KongClient
 
-	PublishService      string
-	WatchNamespaces     []string
-	WatchReferenceGrant bool
+	PublishService       string
+	WatchNamespaces      []string
+	EnableReferenceGrant bool
```

Comment:
Can we add here a comment similar to the one you added in the HTTPRoute controller?